### ISSUE TITLE
[py2py3] Fix potential bug in WMException

### ISSUE THREAD

**Issue by mapellidario** [dmwm/WMCore#10173]
**Impact of the bug**

This bug could affect WMException and all the modules that depend on it that have already been modernized to py2py3, for example `WMConfigCache`.

**Describe the bug**

The line 
https://github.com/dmwm/WMCore/blob/875c3c5a48d0d82df947cdf13466c469c33ca27a/src/python/WMCore/WMException.py#L32
will always return `False`  in python2.

This is because we import ` from builtins import bytes ` and `bytes` actually is `future.types.newbytes.newbytes` in python2. While `isinstance(message, bytes)` correctly returns `True` when `message` is a python2 `str` (remember that a py2 `str` is a sequence of bytes), if we use `type(message) == bytes` this will try to match `str` with `future.types.newbytes.newbytes` and will always return `False`.

Effect: when formatting an exception message containing non-ascii characters from a modernized module that imports `str()` from future, WMException fails with 

```plaintext
Traceback (most recent call last):
  File "/home/dmwm/wmcore_unittest/WMCore/test/python/WMCore_t/WMException_t.py", line 62, in testExceptionUnicodeAsBytesSequence
    self.logger.debug("String version of exception: " + str(exception))
  File "/home/dmwm/unittestdeploy/wmagent/1.4.0-comp/sw/slc7_amd64_gcc630/external/py2-future/0.18.2/lib/python2.7/site-packages/future/types/newstr.py", line 104, in __new__
    return super(newstr, cls).__new__(cls, value)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/WMException.py", line 149, in __str__
    strg += "Message: %s\n" % self._message
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 74: ordinal not in range(128)
```

**How to reproduce it**

Add `from builtins import str` at the beginning of `test/python/WMCore_t/WMException_t.py` and run those unit tests.

**Additional context and error message**

`from builtins import str` is not at the beginning of `test/python/WMCore_t/WMException_t.py` yet because I modernized the unit test in #9868 with the goal "make WMException work", while the proper goal should have been "make WMException work, while calling it from a properly modernized unit test as if it were a generic WMCore module". This would have forced me to update `str()` in
https://github.com/dmwm/WMCore/blob/875c3c5a48d0d82df947cdf13466c469c33ca27a/test/python/WMCore_t/WMException_t.py#L67
to the modernized `str()` provided from future.

The possibility that this bug could happen emerged when investigating failing unit tests in #10172 

**Update** after opening #10187 : the correct purpose should have been "make WMException work, while calling it from a properly modernized unit test **and** from a non-modernized unit test". Pursuing the incorrect purpose caused #10174 to be inadequate to solve this issue and #10187 was necessary

**Expected behavior**

We should properly handle strings in WMException, and update WMException_t accordingly.

**Comment by amaltaro** [comment]
I guess this explains why we started seeing such errors. One of them have been recently spotted and fixed in:
https://github.com/dmwm/WMCore/pull/10152

Thanks for investigating it, Dario.

**Comment by mapellidario** [comment]
Hello @amaltaro , I am sorry. I have to re-open this issue. I realized only after you merged #10174 that it would help us with modernized code, but it did not help us with non-modernized code, such as the bug that you saw in #10152. I have a new PR for addressing non-modernized code
